File: src/main/java/br/anhembi/funmodechild/repository/OrderRepository.java
package br.anhembi.funmodechild.repository;

import br.anhembi.funmodechild.entity.Order;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface OrderRepository extends MongoRepository<Order, String> {

    List<Order> findByCustomerId(String customerId);

    Order findByIdAndCustomerId(String id, String customerId);
}

File: src/main/kotlin/br/anhembi/funmodechild/repository/OrderRepository.kt
package br.anhembi.funmodechild.repository

import br.anhembi.funmodechild.entity.Order
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface OrderRepository : MongoRepository<Order, String> {
    fun findByCustomerId(customerId: String): List<Order>

    fun findByIdAndCustomerId(id: String, customerId: String): Order?
}
